cityofaustin/atd-moped
# Activity Log enhancements

## Associated issues

~~This is my first stab at the way I want to refactor the activity log, by splitting the different activity updates into their own components. That way it will be easier to debug (i hope) and fine tune behaviors based on which table the update is coming from.~~

* I removed all the code for the getLookups hooks, its not working in production. Also I think instead of doing a lot of work to check which lookup tables need to be queried, let's just query the lookup tables. Eventually I'd like to investigate moving them into Context app wide, since I think we look them up in a few places. 


~~While this PR is ready to review, the whole activity log is still a work in progress, I am going to chip away at all the tables and make new components for them as I update their event hooks to be the rest connector ones.~~

Thanks to feedback and help from John, I've restructured it again. Instead of each table/activity type getting its own component, there is one activity component and multiple functions to format the information that should be rendered. 

Also (thanks again to John), I dropped the "from" part of the activity log text to only say "[thing] was updated to [update]" instead of including. 

Stylistically and textwise it needs improving but this is a first pass to clean things up. 

## Testing
**URL to test:** 
https://deploy-preview-910--atd-moped-main.netlify.app/moped/projects/1905?tab=activity_log


**Steps to test:**

Check the activity log, since this is a preview app it points to staging, so there is a mix of legacy (pre REST connector) activity log events and also the new ones (moped project and moped tags). Updating tags and updating the project summary fields will add new activity entries. 


---
#### Ship list
